python/tskit/ms.py

```diff
@@ -0,0 +1,129 @@
+        """
+        breakpoints = list(self._tree_sequence.breakpoints(True)) + [self._num_loci]
+        if self._num_loci == 1:
+            tree = next(self._tree_sequence.trees())
+            newick = tree.newick(precision=self._precision)
+            print(newick, file=output)
+        else:
+            j = 1
+            for tree in self._tree_sequence.trees():
+                newick = tree.newick(precision=self._precision)
+                left, right = tree.interval
+                while j < (len(breakpoints) - 1) and breakpoints[j] <= right:
+                    length = breakpoints[j] - left
+                    left = breakpoints[j]
+                    j += 1
+                    # Print these seperately to avoid the cost of creating
+                    # another string.
+                    print("[{}]".format(int(length)), end="", file=output)
+                    print(newick, file=output)
+
+    def __write_header(self, output):
+        print(
+            "ms {} {} # This file is an ms-style output file generated from tskit. The two arguments written are sample size and number of replicates".format(
+                self._tree_sequence.sample_size, max(self._num_replicates, 1)
+            ),
+            file=output
+        )
+        print("{} # Setting random seed to 999".format(999), file=output)
+
+    def write(self, output):
+
+        if os.path.getsize(output.fileno()) == 0:
+            self.__write_header(output)
+
+        tree_sequence = self._tree_sequence
+        print(file=output)
+        print("//", file=output)
+        if self._print_trees:
+            self.print_trees(output)
+        if self._mutation_rate > 0:
+            s = tree_sequence.get_num_sites() 
+            print("segsites:", s, file=output)
+            if s != 0:
+                print("positions: ", end="", file=output)
+                positions = [
+                    variant.position / (tree_sequence.sequence_length)
+                    for variant in tree_sequence.variants()
+                ]
+                positions.sort()
+                for position in positions:
+                    print(
+                        "{0:.{1}f}".format(position, self._precision),
+                        end=" ",
+                        file=output,
+                    )
+                print(file=output)
+                for h in tree_sequence.haplotypes():
+                    if set(h).issubset({'0', '1', '-'}):
+                        print(h, file=output)
+                    #####################################################################################
+                    # Introducing an error to exit if the sequence is not compatible with the ms format #
+                    #####################################################################################
+                    else:
+                        sys.exit("This tree sequence contains non-biallelic SNPs and is incompatible with the ms format!")
```

Comment:
An exception is more appropriate here, .e.g. ``raise ValueError("not compatible with ms output")``